mongoengine/connection.py

```diff
@@ -264,6 +267,8 @@ def disconnect(alias=DEFAULT_CONNECTION_NAME):
     if alias in _connection_settings:
         del _connection_settings[alias]
 
+    _local_sessions.clear_all()
```

Comment:
Yeah...in retrospect I think I was trying to protect from the very weird use case of: someone calls `disconnect` inside of a transaction.

This is definitely over board for that use case. I think, instead, it's OK to say "_don't call disconnect in a transaction - this will result in undefined behavior_"